metal/v1/api_href.go

```diff
@@ -0,0 +1,92 @@
+package v1
```

Comment:
I've wondered about where we might best place non-generated helpers, I'm wondering if a separate package (similar to metal-java) would have any benefits or drawbacks compared to this single package approach.

Reply:
href hydration has been moved to https://github.com/equinix-labs/metal-go/pull/84
moving that hydration to a separate package can be investigated there if/when we invest more in supporting that use case